policy/manager.go

```diff
@@ -80,7 +79,25 @@ LOOP:
 		case err := <-policyIDsErrCh:
 			m.log.Error("encountered an error monitoring policy IDs", "error", err)
 			if isUnrecoverableError(err) {
-				break LOOP
+
+				// If we reach this point it means an unrecoverable error happened.
+				// We should reset any internal state and re-run the policy manager.
+				m.log.Debug("re-starting policy manager")
+
+				// Explicitly stop the handlers and cancel the monitoring context instead
+				// of relying on the deferred statements, otherwise the next iteration of
+				// m.Run would be executed before they are complete.
+				m.stopHandlers()
+				cancel()
+
+				// Make sure we start the next iteration with an empty map of handlers.
+				m.lock.Lock()
+				m.handlers = make(map[PolicyID]*Handler)
+				m.lock.Unlock()
+
+				// Delay the next iteration of m.Run to avoid re-runs to start too often.
+				time.Sleep(10 * time.Second)
+				go m.Run(ctx, evalCh)
 			}
 			continue
```

Comment:
The previous question was for here, not for the linter in the test jejejej ups

Reply:
Ahh I see 😄 

We could go that route, and I'm usually for failing loudly, but in this case it would assume the autoscaler to be running in an orchestrated environment (which is a fair assumption), but could also cause scaling actions to fail or be cancelled mid-action, and this actions can take a while to finish.

Since the workers and the policy manager work independently, safely exiting here would require all scaling actions to be paused and wait for those in-flight to continue. During this time any policy change would be ignored, and we would not execute new scaling actions.

Reply:
ah, its about maintaining the current actions , got it